src/Cache/MemoryCacheItemPool.php

```diff
@@ -55,12 +55,18 @@ public function getItem($key)
      *   key is not found. However, if no keys are specified then an empty
      *   traversable MUST be returned instead.
      */
-    public function getItems(array $keys = [])
+    public function getItems(array $keys = []): iterable
     {
         $items = [];
 
-        foreach ($keys as $key) {
-            $items[$key] = $this->hasItem($key) ? clone $this->items[$key] : new Item($key);
+        if (\PHP_VERSION_ID >= 80000) {
+            foreach ($keys as $key) {
+                $items[$key] = $this->hasItem($key) ? clone $this->items[$key] : new TypedItem($key);
+            }
+        } else {
+            foreach ($keys as $key) {
+                $items[$key] = $this->hasItem($key) ? clone $this->items[$key] : new Item($key);
+            }
         }
```

Comment:
I like how explicit it is with two distinct branches, but could it be collapsed into one loop with a variable "false" condition type?

```
$default = \PHP_VERSION_ID >= 80000 ? TypedItem:class : Item:class;
for ($keys as $key) {
  $items[$key] = $this->hasItem($key) ? clone $this->items[$key] : new $default($key);
}
```

Reply:
Done!